### ISSUE TITLE
Intellij reports Example struct doesn't implement GDClass interface in `/test`

### ISSUE THREAD

**Issue by mrwormhole** [godot-go/godot-go#53]
![image](https://user-images.githubusercontent.com/22800416/194644695-fd5a8a71-0d79-4849-a79b-5acc03887575.png)

/test folder actually builds with `make build` but Go file complains but GDExtensions? could this be an issue?

I was just trying it out so I am not sure if this is an issue but `make build` builds so I am a little confused.

**Comment by mrwormhole** [comment]
`make ci_gen_test_project_files` gives this error from the latest (could this be possible breaking change from the mainstream?)
![image](https://user-images.githubusercontent.com/22800416/194665337-b1f9c0d2-7969-404c-9799-7740dcabddb4.png)


![image](https://user-images.githubusercontent.com/22800416/194665204-2b47bda6-22c0-4437-b048-fb53997033de.png)

**Comment by pcting** [comment]
it looks like the project isn't initialized. you need at minimum this fine:
```
godot-go$ cat test/demo/.godot/extension_list.cfg 
res://example.gdextension
```

this cfg file tells godot to load that extension file. try running this:

```
$ make ci_gen_test_project_files
```

then run the tests

```
$ make test
```

**Comment by pcting** [comment]
@mrwormhole any updates?

**Comment by mrwormhole** [comment]
@pcting make build works, make test works without crash.

ci_gen_test_project_files crashes and I can still see example.go is screaming about unsatisfied interface methods. How can I implement the new methods so that example.go doesn't yell at me for being invalid? do we have an estimate around 4.0 stabilization so that we can do a roadmap from there? godot 4.0 is on beta right? it would be really good if they pushed out RC so that no breaking changes would happen and we would put some time on making the dev ex more Go idiomatic.

godot-rust guys are not even touching gdextension integration for the moment(I guess it is because of being flaky), I guess there are lots of breaking changes from gdnative to gdextension. Would you be still maintaining gdnative API for anything below 4.0?

**Comment by pcting** [comment]
> @pcting make build works, make test works without crash.

hurray! thanks for confirming that someone besides me has successfully run the code :+1: 

> ci_gen_test_project_files crashes and I can still see example.go is screaming about unsatisfied interface methods. How can I implement the new methods so that example.go doesn't yell at me for being invalid? 

What IDE are you using? Intellij? i'll look into why the interface isn't being satisfied. I'm using vscode and I don't think I've seen that issue.

> do we have an estimate around officially Godot's 4.0 stabilization so that we can do a roadmap from there? godot 4.0 is on beta right? it would be really good if they pushed out RC so that no breaking changes would happen and we would put some time on making the dev ex more Go idiomatic.

i think once godot 4 goes to RC, we'll see the interface stabilize. it'll definitely be nice of the godot team to stabilize the interface, but I imagine they have more important things to deal with.

--

my goal overall goal:
* reach feature parity with godot-cpp.
* design godot-go so that users are writing pretty much idiomatic go.

right now i'm working towards implementing a real world demo app to find gaps in the current API.

**Comment by mrwormhole** [comment]
yes I am using intelliJ based IDE, it is quite odd that it is getting compiled if interface is not being satisfied, there is probably wrong in my setup then

**Comment by mrwormhole** [comment]
can't reproduce in 2026, will close
